Text handling doc change (#2350)
openvinotoolkit/model_server: docs/text_handling.md

We addressed both the situation when the original model requires tokens on input or output and there is added support for models with embedded tokenization layer. Below are demonstrated use cases with a simple client application sending and receiving text in a string format. Whole complexity of the text conversion is fully delegated to the remote serving endpoint.

> ## Accepting strings by serving Python script

Reviewer: ```suggestion
## Serving a pipeline with tokenization and detokenization delegated to a python node
```

Author: I don't think it is a good title, since strings can be used for other things than tokenization and detokenization when serving python script. @dtrawins

> When the model server is configured to serve python script (via MediaPipe Graph with PythonExecutorCalculator), it is possible to communicate via arbitrary format including string or list of strings. Refer to full [Python execution documentation](python_support/reference.md) or end-to-end [LLM text generation demo](../demos/python_demos/llm_text_generation/README.md).

Reviewer: ```suggestion
When the model server is configured to serve python script (via [MediaPipe Graph](mediapipe.md) with PythonExecutorCalculator), it is possible to communicate between the client and server via messages with a raw bytes content. Both the client and server side script can convert the messages from bytes to strings.

Refer to full [Python execution documentation](python_support/reference.md) or end-to-end [LLM text generation demo](../demos/python_demos/llm_text_generation/README.md).
```

> When the model server is configured to serve python script (via MediaPipe Graph with PythonExecutorCalculator), it is possible to communicate via arbitrary format including string or list of strings. Refer to full [Python execution documentation](python_support/reference.md) or end-to-end [LLM text generation demo](../demos/python_demos/llm_text_generation/README.md).

Reviewer: ```suggestion
When the model server is configured to serve python script (via MediaPipe Graph with PythonExecutorCalculator), it is possible to send a string or a list of strings. Refer to full [Python execution documentation](python_support/reference.md) or end-to-end [LLM text generation demo](../demos/python_demos/llm_text_generation/README.md).
```

> When the model server is configured to serve python script (via MediaPipe Graph with PythonCalculator), it is possible to communicate with arbitrary format conforming Python's buffer protocol including string or list of strings. Refer to full [Python integration documentation](python_support/quickstart.md) or end-to-end [LLM text generation demo](../demos/python_demos/llm_text_generation/README.md).

Reviewer 2: 1. Please use a full name: PythonExecutorCalculator.
2. Mentioning buffer protocol seems like too much of a detail on this level. Maybe it would be better to just say that it's possible to work with encoded strings? 
3. [Python integration documentation] - maybe let's stick with Python "execution" rather than "integration" (we don't use "Python integration" in other parts of the documentation)
4. You are linking to quickstart not to the full documentation.

Author: Done

> ## Custom CPU Extension (Custom Operations) for tokenization via OpenVINO model

Reviewer: ```suggestion
## Serving a model with a string in input layer
```

> Some AI model training frameworks support the layers accepting the string format on the input or output. They include the layers performing the tokenization operations inside the neural network.

Reviewer: ```suggestion
Some AI models support the layers with string format on the input or output. They include the layers performing the tokenization operations inside the neural network.
```

> While core OpenVINO does not contain the layers accepting string data type, it is possible to extend it with a CPU extension.

Reviewer: ```suggestion
OpenVINO supports such layers with string data type using a CPU extension.
```

> While core OpenVINO does not contain the layers accepting string data type, it is possible to extend it with a CPU extension.

Reviewer 2: It's a wider topic, but should we still refer to that as CPU extension?

Author: Unfortunately it still is a cpu extension

Reviewer 2: I mean I can only see "CPU extension" in model server docs. In OpenVINO it's referred to as "custom operations". I wonder if this "CPU extension" name is historical leftover.

Author: I agree, and unfortunately it is --cpu_extension in OVMS parameters as well. I think we should stick to this name due to the param name

Model Server includes a built-in extension for wide list of custom [tokenizers and detokenizers](https://github.com/openvinotoolkit/openvino_contrib/tree/master/modules/custom_operations/user_ie_extensions/tokenizer) layers.

> The extension enables accepting OpenVINO tensors of type `ov::element::Type_t::string` and performing tokenization. The server creates string tensor out of KServe/TensorflowServing API strings automatically.

Reviewer: ```suggestion
The extension performs tokenization operation for the string data type. 
```

A demonstration of such use case is in the MUSE model which can be imported directly but the models server. The client can send the text data without any preprocessing and take advantage of much faster execution time.
Check the [MUSE demo](../demos/universal-sentence-encoder/README.md).

> ## DAG pipeline to delegate tokenization to the server

Reviewer: ```suggestion
## DAG pipeline to delegate tokenization to the server (deprecated)
```

Author: Are we going to remove 2d u8? This will remove any support for sending strings to custom nodes.

Author: Will invalidate blingfire-based custom node we release.
